packages/flutter/lib/src/widgets/conduit.dart

```diff
@@ -0,0 +1,375 @@
+    );
+    if (child == renderObject.child) {
+      assert(child?.parent != null);
+      final _RenderLeaderProxyBox oldRenderObject = renderObject;
+      assert(slot == null);
+      assert(renderObject.child == child);
+      oldRenderObject.child = null;
+      assert(oldRenderObject == renderObject);
+    } else {
+      final AnchorKey key = _previousWidget?.remoteKey ?? widget?.remoteKey;
+      assert(key != null);
+      key.currentContext.removeChildRenderObject(child);
+    }
+  }
+}
+
+/// A widget keyed by an [AnchorKey] and allows a [Conduit] possesses the same
+/// [AnchorKey] to attach the render object of its `remoteChild` to.
+class RenderConduitAnchorWidget extends LeafRenderObjectWidget {
+  /// Creates a [RenderConduitAnchorWidget] with an [AnchorKey].
+  const RenderConduitAnchorWidget({
+    @required AnchorKey key,
+  }) : assert(key != null),
+       super(key: key);
+
+  @override
+  AnchorKey get key => super.key as AnchorKey;
+
+  @override
+  LeafRenderObjectElement createElement() => _RenderConduitAnchorElement(this);
+
+  @override
+  RenderObject createRenderObject(BuildContext context) => _RenderFollowerProxyBox();
+}
+
+class _RenderConduitAnchorElement extends LeafRenderObjectElement {
+  _RenderConduitAnchorElement(RenderConduitAnchorWidget widget) : super(widget);
+
+  @override
+  RenderConduitAnchorWidget get widget => super.widget as RenderConduitAnchorWidget;
+
+  @override
+  _RenderFollowerProxyBox get renderObject => super.renderObject as _RenderFollowerProxyBox;
+
+  @override
+  void mount(Element parent, dynamic newSlot) {
+    super.mount(parent, newSlot);
+
+    final _PendingRenderObjectManifest pending = widget.key._manifest.content;
+
+    if (pending == null)
+      return;
+
+    renderObject.leader = pending.leaderRenderBox;
+    final RenderBox pendingRenderBox = pending.renderObject as RenderBox;
+    insertChildRenderObject(pendingRenderBox, null);
+    widget.key._manifest.content = null;
+  }
+
+  @override
+  void unmount() {
+    renderObject.leader = null;
+    super.unmount();
+  }
+
+  @override
+  void insertChildRenderObject(RenderBox child, dynamic slot) {
+    assert(slot == null);
+    final RenderProxyBox oldRenderObject = renderObject;
+    assert(oldRenderObject.debugValidateChild(child));
+    renderObject.child = child;
+    assert(renderObject == oldRenderObject);
+  }
+
+  @override
+  void removeChildRenderObject(RenderObject child) {
+    final RenderProxyBox oldRenderObject = renderObject;
+    // This is more forgiving than a typical removeChildRenderObject implementation,
+    // since _RenderConduitAnchorElement detaches its remoteChild.renderObject in
+    // `Element.deactivate()`.
+    if (oldRenderObject.child != null) {
+      assert(oldRenderObject.child == child);
+      renderObject.child = null;
+    }
+    renderObject.leader = null;
+    assert(renderObject == oldRenderObject);
```

Comment:
I'm confused by the name "oldRenderObject" here, especially given the final assert. Can we just change the type of the argument? It's already marked covariant in the superclass.

Reply:
I think I followed this but changed the name of the variable:
https://github.com/flutter/flutter/blob/505af78ae153520df6268360d450bf7c980f420f/packages/flutter/lib/src/widgets/framework.dart#L5535-L5542